lms/djangoapps/courseware/date_summary.py

```diff
@@ -213,6 +213,24 @@ def link(self):
             return ecommerce_service.checkout_page_url(course_mode.sku)
         return reverse('verify_student_upgrade_and_verify', args=(self.course.id,))
 
+    @property
+    def is_enabled(self):
+        """
+        Whether or not this summary block should be shown.
+
+        By default, the summary is only shown if it has date and the date is in the
+        future and the user's enrollment is in upsell modes
+        """
+        is_enabled = super(VerifiedUpgradeDeadlineDate, self).is_enabled
```

Comment:
if `is_enabled` is False, let's return early so we don't have to do the database call in the next line